src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -425,52 +511,127 @@ public class SqlMutationEngine : IMutationEngine
+        /// <returns>Returns a FindRequestContext object constructed from the existing context and create/upsert operation results.</returns>
+        private FindRequestContext ConstructFindRequestContext(RestRequestContext context, DbResultSetRow mutationResultRow, string roleName)
+        {
+            FindRequestContext findRequestContext = new(entityName: context.EntityName, dbo: context.DatabaseObject, isList: false);
+
+            // PrimaryKeyValuePairs in the context is populated using the primary key values from the
+            // results of the insert/upsert database operation.
+            foreach (string primarykey in context.DatabaseObject.SourceDefinition.PrimaryKey)
+            {
+                _sqlMetadataProvider.TryGetBackingColumn(context.EntityName, primarykey, out string? backingColumnName);
+                if (!string.IsNullOrEmpty(backingColumnName))
+                {
+                    findRequestContext.PrimaryKeyValuePairs.Add(backingColumnName, value: mutationResultRow.Columns[primarykey]!);
+                }
+                else
+                {
+                    findRequestContext.PrimaryKeyValuePairs.Add(primarykey, value: mutationResultRow.Columns[primarykey]!);
+                }
+            }
+
+            // READ action for the given role can have include and exclude fields configured. Populating UpdateReturnFields
+            // ensures that the select query retrieves only those fields that are allowed for the given role.
+            findRequestContext.UpdateReturnFields(_authorizationResolver.GetAllowedExposedColumns(context.EntityName, roleName, EntityActionOperation.Read));
+
+            return findRequestContext;
+        }
+
+        /// <summary>
+        /// Constructs and returns a HTTP 201 Created response.
+        /// The response is constructed using results of the upsert database operation when database policy is not defined for the read permission.
+        /// If database policy is defined, the results of the subsequent select statement is used for constructing the response.
+        /// </summary>
+        /// <param name="resultRow">Reuslt of the upsert database operation</param>
+        /// <param name="jsonDocument">Result of the select database operation</param>
+        /// <param name="isReadPermissionConfiguredForRole">Indicates whether read permissions is configured for the role</param>
+        /// <param name="isDatabasePolicyDefinedForReadAction">Indicates whether database policy is configured for read action</param>
+        private static CreatedResult ConstructCreatedResultResponse(Dictionary<string, object?> resultRow, JsonDocument? jsonDocument, string primaryKeyRoute, bool isReadPermissionConfiguredForRole, bool isDatabasePolicyDefinedForReadAction)
+        {
+            // When the database policy is defined for the read action, a subsequent select query will be executed to fetch the results.
```

Comment:
Instead of mentioning this as a subsequent select query (which can be confused with the subsequent select that we perform to fetch results using a subsequent select for trigger - but within the same db roundtrip), it would be more appropriate to mention that when read action is defined, we perform another db roundtrip to get back the data.

Reply:
Added